### ISSUE TITLE
FileFilter Do Not Affects In File.browseForOpen(..) Operation

### ISSUE THREAD

**Issue by rat-moonshine** [openfl/openfl#2600]
**Describe the bug**
**Output Platform:** macOS
**Tested On:** macOS Monterey

`File.browseForOpen(..)` ignores provided `FileFilter` objects during its operation. Targeted file-type remains disabled/non-selectable during a file browsing operation.

**To Reproduce**

Demo Code:

```
private function openFile():Void
{
	var pathCheckingFile = new File();
	pathCheckingFile.addEventListener(Event.SELECT, onFileSelectHandler);
        pathCheckingFile.addEventListener(Event.CANCEL, onFileSelectCancelHandler);
        pathCheckingFile.browseForOpen("Open Text File", [new FileFilter("Open", "*.athx")]); // simple text file with different extension
}
```

**Expected behavior**
File browsing provide ability to select the file-types those provided in `FileFilter` objects.

**OpenFL Targets**
Openfl [git-9.3.0-Dev]

**Comment by dimensionscape** [comment]
```hx
file.browseForOpen("Open", [new FileFilter("File", "ext")]);
```

Remove the `*.` from your extension string and it should work as expected. Let me know if you have any further issue with this.

**Comment by rat-moonshine** [comment]
Hey @Dimensionscape , thank you, for a quick response. 

I tried by the way you suggested, but I still see targeted file-type is disabled in file-browser:
`pathFile.browseForOpen("Open Text File", [new FileFilter("Open", "athx")]);`

![image](https://user-images.githubusercontent.com/24370870/198009126-eed625f5-fa3a-413c-87d9-8e95e0238730.png)

On a side note, the [FileFilter](https://api.openfl.org/openfl/net/FileFilter.html) API suggests to use `*.` in the documentation.

**Comment by dimensionscape** [comment]
> Hey @Dimensionscape , thank you, for a quick response.
> 
> I tried by the way you suggested, but I still see targeted file-type is disabled in file-browser: `pathFile.browseForOpen("Open Text File", [new FileFilter("Open", "athx")]);`
> 
> ![image](https://user-images.githubusercontent.com/24370870/198009126-eed625f5-fa3a-413c-87d9-8e95e0238730.png)
> 
> On a note, the [FileFilter](https://api.openfl.org/openfl/net/FileFilter.html) API suggests to use `*.` in the documentation.

You're correct `*.` should work based on the flash docs but it was omitted in the recent implementation by mistake. I'm fixing that now for the next patch release, however, it should work without it. There may be a lower level bug in our native file dialog lib or an incompatibility with newer versions of the macOS that needs to be looked at since it works for Windows. I'll look into it more.

**Comment by dimensionscape** [comment]
```hx
@:noCompletion private function __getFilterTypes(typeFilter:Array<FileFilter>):String
	{
		var filter:String = null;

		if (typeFilter != null)
		{
			var filters = [];

			for (type in typeFilter)
			{
				filters.push(StringTools.replace(StringTools.replace(type.extension, "*.", ""), ";", ","));
			}

			filter = filters.join(";");
		}
		return filter;
	}
```

Actually, after further review, I didn't omit this. It should work with or without the asterisk+period. This points to an issue in Lime or the native library behind our file dialog in Lime.

**Comment by joshtynjala** [comment]
> This points to an issue in Lime or the native library behind our file dialog in Lime.

I've determined that, in Lime's _FileDialog.cpp_, the methods `FileDialog::OpenFile`, `FileDialog::OpenFiles`, and `FileDialog::SaveFile` all assume that the filter contains a single file extension only. The functions from tinyfiledialogs that we're using can actually accept more than one file extension, though.

While my memories of C++ have mostly faded away, I eventually came up with a fix for non-Windows platforms today. I'll try to get it working on Windows soon, and then I can commit the fix to Lime.
